src/network/networkd-network-gperf.gperf

```diff
@@ -178,6 +178,9 @@ IPv6Prefix.OnLink,                      config_parse_prefix_flags,
 IPv6Prefix.AddressAutoconfiguration,    config_parse_prefix_flags,                      0,                             0
 IPv6Prefix.ValidLifetimeSec,            config_parse_prefix_lifetime,                   0,                             0
 IPv6Prefix.PreferredLifetimeSec,        config_parse_prefix_lifetime,                   0,                             0
+CAN.BitRate,                            config_parse_si_size,                           0,                             offsetof(Network, can_bitrate)
+CAN.SamplePoint,                        config_parse_uint32,                            0,                             offsetof(Network, can_sample_point)
```

Comment:
hmm, config_parse_uint32 is a bit sloppy here, no? I mean, i presume the SamplePoint= field only takes a value in the range 0…1000 or so? please add a more precise parser here...

also, iiuc this is a permille value, right? if so, please consider adding parse_permille() like we already have parse_percent(), and then match this up with config_parse_permille() too...

parse_permille() should probably accept values such as "55‰". But given that that involves unicode, an alternative syntax with a single separating dot followed by a single digit might make sense to support as well, i.e. "5.5%"?

i.e. something along the lines of:

```c
int parse_permille_unbounded(const char *p) {
        const char *pc, *pm, *n;
        int r, v;

        pm = endswith(p, "‰");
        if (pm) {
                n = strndupa(p, pm - p);
                r = safe_atoi(n, &v);
                if (r < 0)
                        return r;
        } else {
                pc = endswith(p, "%");
                if (!pc)
                        return -EINVAL;

                dot = memchr(p, pc - p, '.');
                if (dot) {
                        if (dot + 2 != pc)
                                  return -EINVAL;
                        if (dot[1] < '0' || dot[1] > '9')
                                  return -EINVAL;
                        q = dot[1] - '0';
                        n = strndupa(p, dot - p);
                } else {
                        q = 0;
                        n = strndupa(p, pc - p);
                }
                r = safe_atoi(n, &v);
                if (r < 0)
                        return r;
                if (v * 10 + q < v) /* overflow */
                        return -ERANGE;

                v = v * 10 + q;
        }
        if (v < 0)
                return -ERANGE;

        return v;
}

int parse_permille(const char *p) {
        int v;

        v = parse_permilled_unbounded(p);
        if (v > 1000)
                return -ERANGE;

        return v;
}
```

(this is entirely untested, but you get the idea)

Reply:
Works now, I've also added some test.